src/server/server.ts

```diff
@@ -199,20 +202,20 @@ export class GraphQLTransportWSServer {
             let promisedParams = Promise.resolve(baseParams);
 
             if (this.onSubscribe) {
-              promisedParams = Promise.resolve(this.onSubscribe(parsedMessage, baseParams, connection));
+              promisedParams = Promise.resolve(this.onSubscribe(parsedMessage, baseParams, connectionContext.socket));
```

Comment:
lets rename to onRequest hook, and add onSubscribe again which will emit warn message for deprecation and return onRequest??